src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -1934,7 +2080,8 @@ pull_var_clause_default(Node *node)
  */
 static MultiNode *
 ApplyJoinRule(MultiNode *leftNode, MultiNode *rightNode, JoinRuleType ruleType,
-			  List *partitionColumnList, JoinType joinType, List *joinClauseList)
+			  List *partitionColumnList, JoinType joinType, List *joinClauseList,
+			  bool passJoinClauseDirectly)
```

Comment:
the passJoinClauseDirectly argument could use more clarification (why is it needed?)